packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -46,47 +49,48 @@ function BlockForType( {
 	wrapperProps,
 	blockWidth,
 } ) {
-	const defaultColors = useEditorFeature( 'color.palette' ) || [];
+	const defaultColors = useEditorFeature( 'color.palette' ) || emptyArray;
+	const globalStyle = useGlobalStyles();
+	const mergedStyle = useMemo( () => {
+		return getMergedGlobalStyles(
+			globalStyle,
+			wrapperProps.style,
+			attributes,
+			defaultColors
+		);
+	}, [
+		defaultColors,
+		globalStyle,
+		JSON.stringify( wrapperProps.style ),
```

Comment:
Could you please leave the comment there why you had to use `JSON.stringify` _(basically I mean something similar to the explanation which you posted in the PR description_)?